### ISSUE TITLE
[bug] Tauri creating AppData\locale\*app*\EBWebView with privliges

### ISSUE THREAD

**Issue by FabioPlunser** [tauri-apps/tauri#7491]
### Describe the bug

Hi, I have a problem with tauri as a priveleged program. 
I added the manifest so that tauri asks for admin permission, but i have the problem that after it starts it says: "Microsoft Edge doesn't have write or read permission: C:\Users\admin\AppData\Local\EBWebView 

now does it really need this folder or can i somehow disable it. 

The Programm needs privelges because it's a small programm that executes a few powershell commands to add a computer to a specific domain and install programms. 

###Edit
The problem is that Microsoft Edge Webview2 does not have elevated privilegeswhen i execute as admin or with a manyfest can i somehow change the config so that webview also starts with privileges

### Reproduction

Write a Tauri app and execute it with admin privliges. But not the Microsoft Administrator but a Domain admin or another use who has admin rights but you don't have them. 

### Expected behavior

I expected it just starts in the scope as Admin as every other program

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.19045 X64
    ✔ WebView2: 115.0.1901.183
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.71.0 (8ede3aae2 2023-07-12)
    ✔ Cargo: 1.71.0 (cfd3bbd8f 2023-06-08)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.16.1
    - pnpm: 8.6.8
    - npm: 9.5.1

[-] Packages
    - tauri [RUST]: 1.4.1
    - tauri-build [RUST]: 1.4.0
    - wry [RUST]: 0.24.3
    - tao [RUST]: 0.16.2
    - @tauri-apps/api [NPM]: 1.4.0
    - @tauri-apps/cli [NPM]: 1.4.0

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: ../build
    - devPath: http://localhost:5173/
    - framework: Svelte
    - bundler: Vite
```

**Comment by FabianLars** [comment]
I think we can solve this by simply _not_ creating the data dir ourselves and leaving it to webview2 to create it. This seems to work on my machine, with and without admin permissions, without any panics we apparently had 2 years ago.

**Comment by FabioPlunser** [comment]
@FabianLars I don't understand your answer I am not creating the data dir. It gets created automatically. 

But with admin permissions, where the admin is another use it tries to create the dir inside appdata of the admin but the webview doesn't have admin access. So i get the error that microsoft edge can't write to the admin directory. 


Somehow the webview has different permission than the tauri app. 

Without admin permissions it just creates in the current logged in user appdata folder. 

Sorry ithink the title of this issue doesn't represent my issue. 

I have the programm installed with the manifest so that it has to be startet with admin permissions. 
I do that on a local admin and join the pc to a domain with the program (the program executes powershell commands to handle domain registration, and handle install of certain programs once in domain for a domain user) 

Than i restart and logg into a user inside the domain. 
Their i give the same admin permissions on ask but i get the error that microsoft edge can't read or write to the local admin.

**Comment by FabianLars** [comment]
This was not meant as an answer or workaround for you but for how to resolve it inside tauri itself.
For you the workaround would be to find a way to create the directory without admin permissions before the `tauri::Builder` line in main.rs but i can't think of a way to do that rn.

**Comment by FabioPlunser** [comment]
Oh ok i misunderstood that.
Is this something i could change myself? 
That the directory doesn't get created by tauri but by webview to test if this would solve my issue?

**Comment by FabianLars** [comment]
First, delete the `C:/Users/<username>/AppData/local/<bundle-identifier>` folder. Then add this to your Cargo.toml file:
```toml
[patch.crates-io]
tauri = { git="https://github.com/tauri-apps/tauri", branch="fix/datadir" }
tauri-build = { git="https://github.com/tauri-apps/tauri", branch="fix/datadir" }
```
Then, run `cargo update` in your src-tauri dir and try again.

cc @FabioPlunser

**Comment by FabioPlunser** [comment]
Ok this doesn't seem to work for me. 

## My Steps 
I log into an domain admin account install the program and execute it. 
Then i switch into another (none admin) domain account and try to execetue the program as admin (because the ps scripts it runs need admin permissions) and I instantly get the error "Microsoft Edge doesn't have read and write permissions to: C:\Users\admin\AppData\Local\TauriApp\EBWebView

**Comment by FabianLars** [comment]
hmm, i can't think of a way to fix this right now other than somehow checking the folder permissions first and creating a second dir next to it as a fallback, or maybe even try to delete the old folder since you start it with admin).

Does the same error happen when you run the app without admin privileges (after you ran it with a domain admin account first) ?

Also, how did you install the app, via the msi or setup.exe or something custom? And in what dir?

Lastly, that non-admin account is not called `admin` right? because then i don't get why it tries to access `C:\Users\admin` in the first place here 🤔

**Comment by FabioPlunser** [comment]
This is a weird Windows thing. Because i start the app as admin but this admin is a another user on the domain, the app runs in the scope of the user admin. 
For example if you execute notepad as admin and you go to file -> open you get the directories and files of the user admin and not the currently logged in user Fabio. 
Because it runs in the scope of anothre user. 

So if i start the App without admin permissions it creates the directory for current logged in user and if I start it with admin rights. 
It tries to create it in that users AppData but because I am logged in with a different user and the app just runs in this scope maybe not all subproccesses that are needed for this app get elevated to this scope and so i get this error message. 
Or because AppData is usually a hidden directory you are just not allowed to read/write to it from any other user no matter the permission. 


I tried both insallers .msi puts it in C\ProgramFiles\ .exe puts it in C:\User\*user*\Appdata

Yeah the non-admin account is called Fabio and because i execute a as admin and the admin as another user with admin permissions it runs it in the scope of the user. 


I think there is nothing more we could do. 
I fixed it by chaning our idea of the program and just do all the configuration and installing of programs inside the admin account without changing to a non-admin account. 


But this adventure certaintly is interesting Windows permission handling is really weird.
